src/Kernel32.Shared/Kernel32+SafeObjectHandle.cs

```diff
@@ -37,7 +37,7 @@ public SafeObjectHandle(IntPtr preexistingHandle, bool ownsHandle)
             }
 
             /// <inheritdoc />
-            public override bool IsInvalid => this.handle == INVALID_HANDLE_VALUE;
+            public override bool IsInvalid => this.handle == INVALID_HANDLE_VALUE || this.handle == IntPtr.Zero;
```

Comment:
Makes sense. If you could add a comment to the code explaining that, that would be great. :)